src/pages/index.tsx

```diff
@@ -1,82 +1,382 @@
+    const numServices = shop.services.length;
+    for (let i = 0; i < numServices; i++) {
+      if (shop.services[i]) {
+        const parts = shop.services[i]?.parts;
+        if (parts && parts.length !== 0) {
+          const flag = parts.every((part) => {
+            return (
+              selectedTypeFilters.includes(part.build) &&
+              selectedConditionFilters.includes(part.condition)
+            );
+          });
+          if (flag) {
+            return true;
+          }
+        }
+      }
+    }
+    return false;
+  };
+
+  const onSearch = () => {
+    // Fetch via search parameters
+    if (searchString !== "") {
+      switch (searchFilter) {
+        case "Service":
+          getFilteredShops(searchString, false).then((data) => {
+            if (data) {
+              // Filter by part type here
+              let filteredData = data.filter(filterByParts);
+              setShops(filteredData);
+            }
+          });
+          break;
+        case "Shop Name":
+          getFilteredShops(searchString, true).then((data) => {
+            if (data) {
+              // Filter by part type here
+              let filteredData = data.filter(filterByParts);
+              setShops(filteredData);
+            }
+          });
+          break;
+        default:
+          break;
+      }
+    } else {
+      getFilteredShops("", true).then((data) => {
+        if (data) {
+          let filteredData = data.filter(filterByParts);
+          setShops(filteredData);
+        }
+      });
+    }
+  };
+
+  const shopOnClick = (shop: IShop & { services: IService[] }) => {
+    Router.push(`/shop/${shop.id}`);
+  };
+
+  const generateServiceList = (
+    shop: IShop & { services: IService[] },
+    view: string
+  ) => {
+    const maxChip = view === "desktop" ? MAX_CHIP : MAX_CHIP_MOBILE;
+    let serviceList: JSX.Element[] = [];
+    for (let i = 0; i <= maxChip - 1; i++) {
+      let service = shop.services[i];
+      if (service) {
+        serviceList.push(<Chip className={styles.chip} label={service.name} />);
+      }
+    }
+    if (view === "desktop") {
+      serviceList.push(
+        <Chip className={styles.fullListChip} label="See Full List" />
+      );
+    } else {
+      serviceList.push(<Chip className={styles.fullListChip} label="..." />);
+    }
+
+    return serviceList;
+  };
+
+  const itemTemplate = (
+    shop: IShop & { services: IService[] },
+    view: string
+  ) => {
+    const serviceList = generateServiceList(shop, view);
+
+    return (
+      <div className={styles.itemContainer} onClick={() => shopOnClick(shop)}>
+        <Image
+          src={image}
+          alt={shop.name}
+          width={image.width * 0.4}
+          height={image.height * 0.17}
+        />
+        <div className={styles.itemText}>
+          <h4 className={styles.itemShopName}>{shop.name}</h4>
+          <div>{shop.address}</div>
```

Comment:
Can we show full address here?
```suggestion
          <div>{`${shop.address}, ${shop.city}, ${shop.province}, ${shop.postalCode}`}</div>
```

Reply:
I'll have to reformat mobile if I do full address there, do you want me to update just desktop or update both

Reply:
Can't the address just flow to the next line as it already does:
<img width="359" alt="image" src="https://user-images.githubusercontent.com/34189743/218205484-11dce552-b7a3-43e9-9cc2-bc2503c5a545.png">

Image gets wonky tho (vertically elongated) and close to edge of the container